controllers/ocirepository_controller.go

```diff
@@ -362,6 +364,39 @@ func (r *OCIRepositoryReconciler) reconcileSource(ctx context.Context, obj *sour
 		return sreconcile.ResultEmpty, e
 	}
 
+	// Verify the image
+	if obj.Spec.Verify != nil {
+		provider := obj.Spec.Verify.Provider
+		switch provider {
+		case "cosign":
+			if verified, err := r.verify(ctx, obj, url); err != nil {
+				e := serror.NewGeneric(
+					fmt.Errorf("failed to verify '%s' using provider '%s': %w", url, provider, err),
+					sourcev1.SourceVerifiedFailedReason,
+				)
+				conditions.MarkFalse(obj, sourcev1.SourceVerifiedCondition, e.Reason, e.Err.Error())
+				return sreconcile.ResultEmpty, e
+			} else {
+				if !verified {
+					e := serror.NewGeneric(
+						fmt.Errorf("no matching signatures '%s' using provider '%s': %w", url, provider, err),
+						sourcev1.SourceVerifiedFailedReason,
+					)
+					conditions.MarkFalse(obj, sourcev1.SourceVerifiedCondition, e.Reason, e.Err.Error())
+					return sreconcile.ResultEmpty, nil
+				}
+			}
+		default:
+			err := fmt.Errorf("could not found the given %s provider, only valid provider for verification is: cosign", provider)
+			e := serror.NewGeneric(
+				fmt.Errorf("failed to verify '%s' using provider '%s': %w", url, provider, err),
+				sourcev1.SourceVerifiedFailedReason,
```

Comment:
This is more of an unsupported provider situation which requires a change in the OCIRepo configuration. I believe this should result in a stalled condition and require human intervention.

Reply:
This code could be removed, the Kubernetes API blocks resources with invalid providers due to `+kubebuilder:validation:Enum=cosign`.

Reply:
> This code could be removed, the Kubernetes API blocks resources with invalid providers due to +kubebuilder:validation:Enum=cosign.
Nice point, yeah, thanks.